desktop/lib.go

```diff
@@ -287,16 +287,16 @@ func redeemResellerCode(email, currency, deviceName, resellerCode *C.char) *C.ch
 		ResellerCode:   C.GoString(resellerCode),
 		Provider:       "reseller-code",
 	})
-	log.Debugf("DEBUG: redeeming reseller code response: %v", response)
-	if response.Error != "" {
-		log.Debugf("DEBUG: error while redeeming reseller code reponse is: %v", response.Error)
-		return sendError(errors.New("Error while redeeming reseller code: %v", response.Error))
-	}
```

Comment:
This error check needed to be moved up. Fixes the following issue:

```
Nov 11 18:23:13.844 - 0m46s DEBUG webclient: response.go:66 response body: {"error":"wrong-reseller-code","errorId":"wrong-reseller-code","status":"error"} status code 200
Nov 11 18:23:13.845 - 0m46s DEBUG lantern-client.main: lib.go:290 DEBUG: redeeming reseller code response: <nil>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x140e7efec]

goroutine 66 [running, locked to thread]:
main.redeemResellerCode(0x600001d6d420, 0x600001f7c4f0, 0x600001f72070, 0x600001d6c340)
	github.com/getlantern/lantern-client/desktop/lib.go:291 +0x24c
```

Reply:
Thanks for fixing this.